### ISSUE TITLE
fitNewDocuments - Error in as(x, "matrix")[, j, drop = drop] : subscript out of bounds

### ISSUE THREAD

**Issue by chris-billingham** [bstewart/stm#160]
Hi there,

I'm having some problems getting fitNewDocuments to complete on a dataset.

It seems to get to the end of the process but then throws the error. i've tried a few different transformations of the base data but to no avail. I've debugged in and it seems to be keeling over at this part:

`  lambda <- do.call(rbind, lambda)
  lambda <- cbind(lambda,0)
  theta <- exp(lambda - row.lse(lambda))`

However I cannot for the life of me work out why it's not working.

Is this a bug?

Thanks, Chris

**Comment by bstewart** [comment]
Sorry for the trouble! This might be an issue with matrixStats again? ( #89) Can you send me a sessionInfo() call so I can work out what version and OS you are on?

**Comment by chris-billingham** [comment]
Hey!

Sorry I meant to close this the other day as I worked out the issue but my wife decided to give birth instead so I've been preoccupied.

Turned out the issue was that I had mismatched/NULL factor levels in a the fitNewDocuments data that weren't aligned to the modelling data in the prevalence part. Seems the data I was pulling to score had introduced a rogue space at some point which then messed up my standard processing pipeline.

We live and learn (to pay attention with factors).
